core/services/relay/evm/mercury/v4/reportcodec/report_codec_test.go

```diff
@@ -0,0 +1,163 @@
+package reportcodec
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	v4 "github.com/smartcontractkit/chainlink-common/pkg/types/mercury/v4"
+)
+
+func newValidReportFields() v4.ReportFields {
+	return v4.ReportFields{
+		Timestamp:          242,
+		BenchmarkPrice:     big.NewInt(243),
+		Bid:                big.NewInt(244),
+		Ask:                big.NewInt(245),
+		ValidFromTimestamp: 123,
+		ExpiresAt:          20,
+		LinkFee:            big.NewInt(456),
+		NativeFee:          big.NewInt(457),
+		MarketStatus:       1,
+	}
+}
+
+func Test_ReportCodec_BuildReport(t *testing.T) {
+	r := ReportCodec{}
+
+	t.Run("BuildReport errors on zero values", func(t *testing.T) {
+		_, err := r.BuildReport(v4.ReportFields{})
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "benchmarkPrice may not be nil")
+		assert.Contains(t, err.Error(), "linkFee may not be nil")
+		assert.Contains(t, err.Error(), "nativeFee may not be nil")
```

Comment:
what about marketstatus?

Reply:
The zero value means CLOSED so it is a valid value. Unlike the price fields which are pointer valued `*big.Int`, the market status is a scalar `uint32` so it can't be nil either.